src/js/common/schemaform/definitions/address.js

```diff
@@ -30,14 +48,69 @@ export function schema(currentSchema, isRequired = false) {
   };
 }
 
-export function uiSchema(label = 'Address') {
+export function uiSchema(label = 'Address', useStreet3 = false) {
+  let fieldOrder = ['country', 'street', 'street2', 'street3', 'city', 'state', 'postalCode'];
+  if (!useStreet3) {
+    fieldOrder = fieldOrder.filter(field => field !== 'street3');
+  }
+
   return {
     'ui:title': label,
-    'ui:field': 'address',
+    // 'ui:field': 'address',
     'ui:validations': [
       validateAddress
     ],
-    'ui:order': ['country', 'street', 'street2', 'city', 'state', 'postalCode'],
+    'ui:options': {
+      updateSchema: (formData, addressSchema, addressUiSchema, index, path) => {
+        const schemaUpdate = { properties: addressSchema.properties };
+        const country = _.get(path.concat('country'), formData) || addressSchema.properties.country.default;
+        const isRequired = addressSchema.required.length > 0;
+
+        let stateList;
+        if (country === 'USA') {
+          stateList = usaStates;
+        } else if (country === 'CAN') {
+          stateList = canProvinces;
+        } else if (country === 'MEX') {
+          stateList = mexStates;
+        }
+
+        if (stateList) {
+          // We have a list and it's different, so we need to make schema updates
```

Comment:
Could this / should this be done in a selector? As long as we're already using reselect... :man_shrugging: